src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -63,12 +64,13 @@ public class ParserUtil {
      * trimmed.
      * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
      */
-    public static Index parseIndex(String oneBasedIndex) throws ParseException {
+    public static Optional<Index> parseIndex(String oneBasedIndex) throws ParseException {
+
         String trimmedIndex = oneBasedIndex.trim();
         if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
             throw new ParseException(MESSAGE_INVALID_INDEX);
         }
-        return Index.fromOneBased(Integer.parseInt(trimmedIndex));
+        return Optional.of(Index.fromOneBased(Integer.parseInt(trimmedIndex)));
     }
```

Comment:
I would recommend against wrapping this as an `Optional<Index>`. Not only do all its dependencies have to accommodate this change, but they also have to do a `null` check that ends up throwing the same exception, which isn't ideal. I think you did this change for the polymorphism of `EditPersonCommandParser`, so I added a suggestion there for your reference.